src/app/src/components/FacilityListItemsTableSearch.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useReducer } from 'react';
+import { func, string } from 'prop-types';
+
+import CircularProgress from '@material-ui/core/CircularProgress';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import TextField from '@material-ui/core/TextField';
+import Search from '@material-ui/icons/Search';
+import Close from '@material-ui/icons/Close';
+
+const KEYCODE_ENTER = 13;
+const KEYCODE_ESCAPE = 27;
+
+function reducer(state, action) {
+    switch (action.type) {
+        case 'update':
```

Comment:
Consider setting these cases as constants.

Reply:
Switched back to `useState`. Thought this decoupled the logic from the component too much in this case. I'm sure there's a better use case for it coming in the future.